<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<title>:: Research ::</title>
<link href="defaultstyles.css" rel="stylesheet" media="screen" />
<script type="text/javascript" src="branding.js"></script>
<script type="text/JavaScript" src="http://www.colorado.edu/templates/faculty-template/js/curvycorners.js"></script>

</head>

<body>
<div id="container">
	<!--div id="ucb"><script type="text/javascript">ucbheader();</script></div-->
    <div id="wrapper">
    <div id="banner">
    	<h1>Sachin Pundhir, Ph.D.</h1>
    </div>
    <div id="nav">
    	<ul>
      	<li><a href="index.html">Home</a></li>
            <li><a href="publications.html">Publications</a></li>
            <li><a href="cv.html">CV</a></li>
            <!--li><a href="page3.html">Projects</a></li-->
            <li><a href="research.html">Research</a></li>
            <li><a href="contact.html">Contact</a></li>
        </ul>
    </div>
    <div id="page">
		
        <div id="content">
        
            <div id="contentbar" class="contentbar4">
                <p align="center"><img src="img/pundhir_wordle.png" border=0></p>
        	</div>

            <p>Recent advances in next-generation sequencing (NGS) have posed a unique challenge in terms of the analysis and interpretation of huge amount of data generated by this technology. Computational biology, a research area that lies at the intersection of computer science and biology is making immense contributions in addressing this ‘big data’ challenge. My research focuses in this area, specifically on to develop and/or use computational and statistical techniques to interpret complex biological datasets. Central to my research are the questions as to:</p>

           <p>
            <ol>
                <li>How despite having same genetic code a huge diversity of cell arises?</li>
                <li>How transcriptome (RNA-seq) and epigenetic (ChIP-seq) landscape defines the identity of a cell?</li>
                <li>How alterations in this landscape drive cellular differentiation, and if aberrant to the formation of cancer cells?</li>
            </ol>
            </p>

            <p>
            These are fundamental questions, an exploration of which can lead to better prognosis of disease and personalized medicine. However, they also raise several challenges due to non-coherent nature of the data and the inherently interdisciplinary nature of the problems. 
            My goal is to address these questions by proposing new computational methods that use standardized NGS data formats and robust statistics to capture patterns that are unique to a particular cell state or disease condition, and by maintaining collaborations with biologists for experimental follow-up and validation.</p>

            <p>A complete description of my research projects is available <a href="doc/pundhir_research_statement.pdf">here</a>.</p>

        </div>
	</div>
    </div>
    <!--div id="footer">
    	<script type="text/javascript">ucbfooter();</script>
        &copy; Regents of the University of Colorado
    </div-->
</div>
</body>
</html>
